lib/ansible/modules/net_tools/nsupdate.py

```diff
@@ -172,11 +173,23 @@ class RecordManager(object):
     def __init__(self, module):
         self.module = module
 
-        if module.params['zone'][-1] != '.':
-            self.zone = module.params['zone'] + '.'
+        if module.params['zone'] is None:
+            if module.params['record'][-1] != '.':
+                self.module.fail_json(msg='record must be absolute when omitting zone parameter')
+
+            try:
+                self.zone = dns.resolver.zone_for_name(self.module.params['record']).to_text()
+            except (dns.resolver.NotAbsolute, dns.resolver.NoRootSOA) as e:
```

Comment:
Been able to trigger a `dns.resolver.NotAbsolute`?

What about a `dns.exception.Timeout` for when the local resolver isn't reachable? Or a `dns.resolver.NoNameservers` for when the local resolver fail to respond successfully?

Reply:
No, considering we check for a trailing dot it's probably not needed. I'll add excepts for those exceptions and remove `NotAbsolute`. Are there any other exceptions you want to be handled?

Also, do you want me to squash the commits into a single one for a clean merge?

Reply:
Yeah, I think one squashed commits makes the most sense here.

Reply:
done